test/puppetlabs/http/client/async_plaintext_test.clj

```diff
@@ -352,4 +356,95 @@
                                  {:socket-timeout-milliseconds 2000})]
               (let [response @(common/get client url {:as :text})]
                 (is (= 200 (:status response)))
-                (is (= "Hello, World!" (:body response)))))))))))
\ No newline at end of file
+                (is (= "Hello, World!" (:body response)))))))))))
+
+(defn- build-content-handler
+  [data initial-bytes-read? server-side-failure?]
+  (fn [_]
+    (let [outstream (PipedOutputStream.)
+          instream (PipedInputStream.)
+          _ (.connect instream outstream)
+          outwriter (io/make-writer outstream {})]
+      ;; Return the response immediately and asynchronously stream some data into it
+      (future
+       (.write outwriter data)
+       (.flush outwriter)
+       (if server-side-failure?
+         (throw (Exception. "Server side failure!")))
+       ; Block until the client confirms it has read the first few bytes
+       ; Socket time out ensures we don't block here forever in practice if it does wrong
+       (if initial-bytes-read?
+         (deref initial-bytes-read?))
+       ; Write the last of the data
+       (.write outwriter "xxxx")
+       (.close outwriter))
+      {:status 200
+       :body instream})))
+
+(defn- clojure-streaming-success
+  [data-size opts]
+  (testing (str "clojure streaming success: " data-size " bytes with opts " opts)
+    (let [data (apply str (repeat (- data-size 4) "x"))
+          ;; We check that we can read some bytes before all are transmitted only if:
+          ;; - :unbuffered-stream is enabled
+          ;; - the total traffic returned is more than about 40K to prevent buffering issues
+          ;;   i.e. data-size is large enough and not compressed
+          initial-bytes-read? (if (and (= :unbuffered-stream (:as opts))
```

Comment:
Yes, it's Jetty, I remember battling with this when I originally wrote the test.